src/as2fm/scxml_converter/mock_bt_generator.py

```diff
@@ -0,0 +1,338 @@
+
+        :param condition_id: The ID of the condition node
+        :return: SCXML root for the mock condition
+        """
+        scxml_name = f"MockCondition_{condition_id}"
+        scxml_root = ScxmlRoot(scxml_name)
+
+        # Data model for tracking execution
+        data_model = ScxmlDataModel(
+            [
+                ScxmlData("execution_count", "0", "int32"),
+                ScxmlData("last_result", "'SUCCESS'", "string"),
+                ScxmlData("success_probability", "0.5", "float32"),
+            ]
+        )
+        scxml_root.set_data_model(data_model)
+
+        # Initial state
+        init_state = ScxmlState("init")
+        init_state.add_transition(ScxmlTransition.make_single_target_transition("idle", []))
+        scxml_root.add_state(init_state, initial=True)
+
+        # Idle state - responds to BT ticks
+        idle_state = ScxmlState("idle")
+
+        # BT tick transition - use probabilistic transitions instead of random expressions
+        # Create transition targets with probabilities
+        success_target = ScxmlTransitionTarget(
+            "return_success",
+            probability=0.5,  # Will be replaced with success_probability during conversion
+            body=[
+                ScxmlAssign("execution_count", "execution_count + 1"),
+                ScxmlAssign("last_result", "'SUCCESS'"),
+            ],
+        )
+
+        failure_target = ScxmlTransitionTarget(
+            "return_failure",
+            probability=0.5,  # Will be replaced with 1 - success_probability during conversion
+            body=[
+                ScxmlAssign("execution_count", "execution_count + 1"),
+                ScxmlAssign("last_result", "'FAILURE'"),
+            ],
+        )
+
+        # Create transition with multiple probabilistic targets
+        tick_transition = BtTick([success_target, failure_target])
+        idle_state.add_transition(tick_transition)
+
+        # Add return states
+        return_success_state = ScxmlState("return_success")
+        return_success_state.add_transition(
+            ScxmlTransition.make_single_target_transition(
+                "idle", body=[BtReturnTickStatus("SUCCESS")]
+            )
+        )
+        scxml_root.add_state(return_success_state)
+
+        return_failure_state = ScxmlState("return_failure")
+        return_failure_state.add_transition(
+            ScxmlTransition.make_single_target_transition(
+                "idle", body=[BtReturnTickStatus("FAILURE")]
+            )
+        )
+        scxml_root.add_state(return_failure_state)
+
+        # BT halt transition
+        halt_transition = BtHalt.make_single_target_transition("idle", body=[BtReturnHalted()])
+        idle_state.add_transition(halt_transition)
+
+        scxml_root.add_state(idle_state)
+
+        return scxml_root
+
+    def generate_mock_action_plugin(self, action_id: str) -> ScxmlRoot:
```

Comment:
I see you are using the classes defined in ScxmlEntries to build the mock BT nodes, but I wonder if it would be easier to provide the BT nodes using the SCXML modeling format (XML-like), and load them in the same way as we are doing it for the BT control nodes. This should make it much more readable and easier to maintain.